### PR TITLE
Add Anthropic Polyfills

### PR DESCRIPTION
**Pull request by petersalas**
- Add  `<SystemMessage>` polyfill
- Add `<UserMessage name={...}>` polyfill
- Remove leading space from output

### CODE REVIEW

File: packages/ai-jsx/src/lib/anthropic.tsx [hunk 1]
```diff
@@ -105,33 +105,47 @@ export async function* AnthropicChatModel(
...
-          );
-      }
-    })
+    messageElements
+      .filter(AI.isElement)
+      .flatMap((message) => {
+        if (message.tag === SystemMessage) {
+          return [
+            <UserMessage>For subsequent replies you will adhere to the following instructions: {message}</UserMessage>,
+            <AssistantMessage>Okay, I will do that.</AssistantMessage>,
```
Comment (NickHeiner) [inline packages/ai-jsx/src/lib/anthropic.tsx:114]: heh, does this help? I've experimented with stuff like this in the past but haven't ever rigorously tested it.

File: packages/ai-jsx/src/lib/anthropic.tsx [hunk 2]
```diff
@@ -105,33 +105,47 @@ export async function* AnthropicChatModel(
...
+          ];
+        }
+
+        return message;
+      })
+      .map(async (message) => {
+        switch (message.tag) {
+          case UserMessage:
+            return `${AnthropicSDK.HUMAN_PROMPT}:${message.props.name ? ` (${message.props.name})` : ''} ${await render(
```
Comment (NickHeiner) [inline packages/ai-jsx/src/lib/anthropic.tsx:123]: Can you add an example of using a `name` prop?

File: packages/ai-jsx/src/lib/anthropic.tsx [hunk 3]
```diff
@@ -171,10 +185,18 @@ export async function* AnthropicChatModel(
...
+    if (isFirstResponse && text.length > 0) {
+      isFirstResponse = false;
+      if (text.startsWith(' ')) {
+        text = text.slice(1);
+      }
+    }
+    resultSoFar += text;
     logger.trace({ completion }, 'Got Anthropic stream event');
-    yield completion.completion;
+    yield text;
```
Comment (NickHeiner) [inline packages/ai-jsx/src/lib/anthropic.tsx:199]: heh, nice

**Comment by NickHeiner** [comment]
If we had a better unit testing story, this would be a DELIGHTFUL AND WONDERFUL application of it.

**Review by NickHeiner — APPROVED** [review]

File: packages/ai-jsx/src/lib/anthropic.tsx (hunk 1 above)
Comment (petersalas) [reply packages/ai-jsx/src/lib/anthropic.tsx:114]: It seemed to work well enough for the demos in `examples` that I tried 😂

Comment (NickHeiner) [reply packages/ai-jsx/src/lib/anthropic.tsx:114]: But like, you noticed a perf improvement with it?

Comment (petersalas) [reply packages/ai-jsx/src/lib/anthropic.tsx:114]: As compared to blowing up, yes :P

Langchain IIRC does something like `User: <admin>$SYSTEM_MESSAGE</admin>`, and this seemed to work better than that in the toy examples I tried 🤷

File: packages/ai-jsx/src/lib/anthropic.tsx (hunk 2 above)
Comment (petersalas) [reply packages/ai-jsx/src/lib/anthropic.tsx:123]: The debate example uses it, which is what I was testing against

File: packages/ai-jsx/src/lib/anthropic.tsx (hunk 1 above)
Comment (NickHeiner) [reply packages/ai-jsx/src/lib/anthropic.tsx:114]: I didn't mean compared to blowing up – I meant specifically just including the `AssistantMessage`.

**Pull request merged**